api/v1alpha3/virtualmachine_types.go

```diff
@@ -236,8 +236,44 @@ type VirtualMachineImageRef struct {
 	Name string `json:"name"`
 }
 
+// VirtualMachineCdromSpec describes the desired state of a CD-ROM device.
+type VirtualMachineCdromSpec struct {
+	// +required
+
+	// ImageName is the Kubernetes object name of the VirtualMachineImage or
+	// ClusterVirtualMachineImage resource of type ISO that is used as the
+	// backing for the CD-ROM device.
+	ImageName string `json:"imageName"`
+
+	// +optional
+	// +kubebuilder:validation:Enum=VirtualMachineImage;ClusterVirtualMachineImage
+	// +kubebuilder:default=VirtualMachineImage
+
+	// ImageKind describes the type of image used as the backing for the CD-ROM
+	// device. It may be either a VirtualMachineImage (default value) or a
+	// ClusterVirtualMachineImage.
+	ImageKind string `json:"imageKind,omitempty"`
+
+	// +optional
+
+	// Disconnect indicates whether the CD-ROM device should be disconnected
+	// from the VM. If omitted or set to false, the CD-ROM device is added and
+	// connected to the VM when the VM is powered on.
+	Disconnect bool `json:"connect,omitempty"`
+}
+
 // VirtualMachineSpec defines the desired state of a VirtualMachine.
 type VirtualMachineSpec struct {
+	// +optional
+
+	// Cdrom describes the desired state of the VM's CD-ROM devices.
+	//
+	// For each CD-ROM device, an ISO-type VirtualMachineImage must be specified
+	// as the backing for the CD-ROM device.
+	//
+	// If multiple CD-ROM devices are specified, they will be attached with the
+	// exact order as they are listed in this field/
```

Comment:
* There is a trailing `/` in the last sentence.
* Please consider rewriting the final sentence as `CD-ROM devices are attached in list-order.`